src/OpenLoco/Objects/ObjectIndex.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include "../Core/Span.hpp"
+#include "Object.h"
+#include <vector>
+
+namespace OpenLoco::ObjectManager
+{
+    struct ObjectIndexEntry
+    {
+        ObjectHeader* _header;
+        char* _filename;
+        char* _name;
+
+        static ObjectIndexEntry read(std::byte** ptr);
+    };
+
+    struct ObjIndexPair
+    {
+        int16_t index;
+        ObjectIndexEntry object;
+    };
+
+    uint32_t getNumInstalledObjects();
+
+    std::vector<std::pair<uint32_t, ObjectIndexEntry>> getAvailableObjects(ObjectType type);
+    bool tryInstallObject(const ObjectHeader& object, stdx::span<const uint8_t> data);
```

Comment:
I'm not too sure if this should be here.

Reply:
Installing it seems like a manager thing, indeed. Assuming it's calling relevant index functions rather than manipulating the index by itself.

Reply:
It actually doesn't call anything index related apart from one check to ensure the object isn't already in the index. Doesn't even mark the index dirty after installing an object.